src/rust/plugin-execution-sidecar/src/work/analyzer_work_processor.rs

```diff
@@ -148,9 +149,14 @@ impl PluginWorkProcessor for AnalyzerWorkProcessor {
         _plugin_id: Uuid,
         job: ExecutionJob,
     ) -> Result<Self::ProducedMessage, PluginWorkProcessorError> {
+        let request_metadata =
+            RequestMetadata::new(vec![("x-trace-id".to_string(), job.trace_id().to_string())]);
         let run_analyzer_response = self
             .analyzer_client
-            .run_analyzer(RunAnalyzerRequest::new(Update::deserialize(job.data())?))
+            .run_analyzer(
+                RunAnalyzerRequest::new(Update::deserialize(job.data())?),
+                Some(request_metadata),
+            )
```

Comment:
this is the only RPC where I've changed the signature. I'm happy to change every rust-proto grpc signature to include a second arg but it felt like overkill for now?